contracts/zxp/game/GameVault.sol

```diff
@@ -9,35 +9,40 @@ import {IObjectRegistry} from "../interfaces/IObjectRegistry.sol";
 import {ObjectRegistryClient} from "../ObjectRegistryClient.sol";
 import {ISeasons} from "./interfaces/ISeasons.sol";
 
+
 contract GameVault is ERC721Wrapper, ObjectRegistryClient, IGameVault {
     bytes32 internal constant SEASONS = "Seasons";
     mapping(uint id => uint block) public stakedAt;
 
     constructor(
         IERC721 underlyingToken,
         IERC20 _rewardToken,
-        string memory name,
-        string memory symbol,
+        string memory underlyingTokenName,
+        string memory underlyingTokenSymbol,
         IObjectRegistry registry,
         bytes32 game
     )
         ObjectRegistryClient(registry)
-        ERC721(name, symbol)
+        ERC721(underlyingTokenName, underlyingTokenSymbol)
         ERC721Wrapper(underlyingToken)
     {}
 
+    // what is this `id` for? how do we get it? maybe a clearer name?
     function _mint(address to, uint id) internal override {
         stakedAt[id] = block.number;
         super._mint(to, id);
     }
 
     function _burn(uint id) internal override {
+        // moved this for checks-effects pattern to avoid reentrancy
+        uint stakedAt = stakedAt[id];
+        stakedAt[id] = 0;
+
         ISeasons(registry.addressOf(SEASONS)).onUnstake(
             id,
             msg.sender,
-            block.number - stakedAt[id]
+            block.number - stakedAt

```

Comment:
Sorry after looking at this and making the change myself i realized it's not actually a bug or a re-entrancy. You know that there's no re-entrancy because it's not just doing some arbitrary tx to an address, it is specifically calling the Seasons, which does not re-enter (or could if it really wanted to).

So it's a bit of a waste of gas to do this, but it is best practices in general and avoids bugs that might get introduced later so will keep.

Reply:
also this change verbatim causes a compiler error, but i fixed that and added this